### ISSUE TITLE
`node` on WSL1 errors, all major versions

### ISSUE THREAD

**Issue by vladdeSV** [teaxyz/cli#491]
Running on WSL1, `Debian GNU/Linux 11 (bullseye)`.

All versions except 14 seem to share a similar error. Unsure what causes this.

```
$ node^14 --version
error: bad address (os error 14)

$ node^16 --version
/home/bibbe/.tea/nodejs.org/v16.20.0/bin/node: 1: ELF: not found
/home/bibbe/.tea/nodejs.org/v16.20.0/bin/node: 2: : not found
/home/bibbe/.tea/nodejs.org/v16.20.0/bin/node: 3: : not found
/home/bibbe/.tea/nodejs.org/v16.20.0/bin/node: 4:
                                                  

: not found                                                
/home/bibbe/.tea/nodejs.org/v16.20.0/bin/node: 1: Syntax error: word unexpected (expecting ")")

$ node^18 --version
/home/bibbe/.tea/nodejs.org/v18.15.0/bin/node: 1: Syntax error: "(" unexpected

$ node^19 --version
/home/bibbe/.tea/nodejs.org/v19.8.1/bin/node: 1: Syntax error: "(" unexpected
```

**Comment by vladdeSV** [comment]
This is with tea `0.26.0`.

**Comment by jhheider** [comment]
In theory, bullseye should work (we build against buster). To rule out a lot of the normal stuff, what's `uname -a`, `echo $SHELL`, and `$SHELL --version` output? This almost looks like an architecture mis-match, but it's unlikely you're running x86 instead x86-64.

**Comment by vladdeSV** [comment]
```
$ uname -a
Linux bastion 4.4.0-19041-Microsoft #2311-Microsoft Tue Nov 08 17:09:00 PST 2022 x86_64 GNU/Linux

$ echo $SHELL
/bin/bash

$ $SHELL --version
GNU bash, version 5.1.4(1)-release (x86_64-pc-linux-gnu)
Copyright (C) 2020 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>

This is free software; you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
```

**Comment by jhheider** [comment]
Yes, all of that should work. Do `tea node^18 --version` or `/home/bibbe/.tea/nodejs.org/v18.15.0/bin/node --version` work?

**Comment by vladdeSV** [comment]
Different output 🤔

```
$ /home/bibbe/.tea/nodejs.org/v18.15.0/bin/node --version
-bash: /home/bibbe/.tea/nodejs.org/v18.15.0/bin/node: cannot execute binary file: Exec format error

$ tea node^18 --version
/home/bibbe/.tea/nodejs.org/v18.15.0/bin/node: 1: Syntax error: "(" unexpected
```

**Comment by jhheider** [comment]
Hm, interesting that it's `x86_64-pc-linux-gnu` instead of `x86_64-unknown-linux-gnu`. Not likely a problem, but interesting to note.

**Comment by vladdeSV** [comment]
> Different output 🤔

Same for v16, v18, and v19

```
$ /home/bibbe/.tea/nodejs.org/v14/bin/node --version
-bash: /home/bibbe/.tea/nodejs.org/v14.21.3/bin/node: Bad address

$ /home/bibbe/.tea/nodejs.org/v16/bin/node --version
-bash: /home/bibbe/.tea/nodejs.org/v16/bin/node: cannot execute binary file: Exec format error

$ /home/bibbe/.tea/nodejs.org/v18/bin/node --version
-bash: /home/bibbe/.tea/nodejs.org/v18/bin/node: cannot execute binary file: Exec format error

$ /home/bibbe/.tea/nodejs.org/v19/bin/node --version
-bash: /home/bibbe/.tea/nodejs.org/v19/bin/node: cannot execute binary file: Exec format error
```

**Comment by jhheider** [comment]
https://github.com/microsoft/WSL/issues/5554. I thought that kernel looked suspiciously old. Debian Bullseye should be using [5.10](https://en.wikipedia.org/wiki/Debian_version_history).

I don't know that anyone has tried WSL1, so that might be at issue.

**Comment by jhheider** [comment]
Someone has to try it some day. Possibly WSL1 is in little enough use that we can just market it `wontsupport`, but it's good to know _if_ it can be supported.

**Comment by vladdeSV** [comment]
This absolutely solved my problem – I was running WSL1, and upgrading to WSL2 fixed it. Upgrading to WSL2 also fixed #490.

Closing this issue in favor of a separate issue about just WSL1.
